### PR TITLE
fix: Calculate Tax properly

### LINKED ISSUE #7835: Tax is not being calculated properly while ordering tickets
**Describe the bug**
If we buy paid tickets of more than one type, the tax on orders page shows 0.00 for all tickets. There is probably some calculation issue in tax system.

Screenshot from 2021-09-21 19-35-25

### PR DESCRIPTION
**Pull request by daretobedifferent18**
Fixes #7835 

#### Short description of what this resolves:
If we buy paid tickets of more than one type, the tax on orders page shows 0.00 for all tickets. There is probably some calculation issue in tax system.

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The acceptance, integration, unit tests and linter pass locally with my changes 
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: app/models/order.js [hunk 1]
```diff
@@ -37,5 +38,15 @@ export default ModelBase.extend(CustomPrimaryKeyMixin, {
...
-  attendees        : hasMany('attendee')
+  attendees        : hasMany('attendee'),
+
+  taxAmount: computed('amount', 'event.tax.isTaxIncludedInPrice', 'event.tax.rate', function() {
+    const taxType = this.event.get('tax.isTaxIncludedInPrice');
+    const taxRate = this.event.get('tax.rate');
+    if (taxType) {
+      return ((taxRate * this.amount) / (100 + taxRate)).toFixed(2);
+    } else {
+      return ((this.amount) / (1 + taxRate / 100) * (taxRate / 100)).toFixed(2);
```
Comment (maze-runnar) [inline app/models/order.js:49]: Can you share thoght process behind this.

Comment (daretobedifferent18) [reply app/models/order.js:49]: Let x be amount without taxIncluded
So we need to calculate tax which will can be calculated by totalAmount = x + x*taxRate/100;
where ```x * taxRate / 100``` is tax added on top of it which is given by 
```js
totalAmount / ( 1 + taxRate / 100 ) * ( taxRate / 100 )  which gives
(taxRate * totalAmount) / (100 + taxRate)
```

**Comment by maze-runnar** [comment]
ok, so if my ticket price is 100, and the tax rate is 50%. And I have selected tax is included in the price, it is showing 33.33 tax amount price.

**Comment by daretobedifferent18** [comment]
> ok, so if my ticket price is 100, and the tax rate is 50%. And I have selected tax is included in the price, it is showing 33.33 tax amount price.

@maze-runnar it is correct because 100 is ticket price inlcuding tax so actual ticket is of 66.66 + 33.33(tax) which gives 100

**Comment by maze-runnar** [comment]
> > ok, so if my ticket price is 100, and the tax rate is 50%. And I have selected tax is included in the price, it is showing 33.33 tax amount price.
> 
> @maze-runnar it is correct because 100 is ticket price inlcuding tax so actual ticket is of 66.66 + 33.33(tax) which gives 100

yes, it is 🔥. good work.

**Review by maze-runnar — APPROVED** [review]

**Comment by mariobehling** [comment]
This did not solve the issue. Tax is still not showing up e.g. compare here https://test.eventyay.com/e/554ef820

Screenshot from 2021-09-23 05-37-24

**Comment by daretobedifferent18** [comment]
@mariobehling This porblem is not showing neither on localsystem and nor on https://eventyay.com/
I think there is some problem in new changes been merges not properly deplying or maybe not deployed to 
https://eventyay.com/

bcz the changes made in PR #3820 is also not showing (below ScreenShot).
Screenshot from 2021-09-23 11-12-11

I think the problem is in deployment and same goes with this PR.

**Comment by mariobehling** [comment]
@daretobedifferent18 We are not deploying on live server. The server with the latest changes of the dev branch is https://test.eventyay.com.

**Comment by daretobedifferent18** [comment]
@mariobehling ok let me see what is the problem

**Pull request merged**